backend-rust/src/graphql_api/context_ext.rs

```diff
@@ -0,0 +1,23 @@
+//! Simple extension trait for the GraphQL context to make it easier to extract
+//! certain data.
+
+use async_graphql::Context;
+use sqlx::PgPool;
+
+use super::{ApiError, ApiResult, ApiServiceConfig};
+
+pub(crate) trait ContextExt {
```

Comment:
Personally I am not a fan of this approach, it is adding an abstraction with one implementation and makes it look like `pool` and `config` is something which is part of the API of `async_graphql`.
To me, the function calls are more straight forward, and I imagine you would not introduce a module just for holding two functions.

Reply:
I don't think this is self-contained, the only logic in here is the call to `.data()`, the relevant types and the logic for actually setting these values are still defined outside this module.
This logic is also tightly bound to how the `Context<'_>` is set up, so this module is not abstracting anything.

Reply:
@DOBEN What do you think? Do you think this extension trait approach to get `context.pool()` is nice or do you prefer `get_pool(context)` with the free-standing functions?

Reply:
If you ask me, I think we should keep the old implementation so we can focus on getting the remaining parts of the block explorer backend ready.